crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -417,6 +420,28 @@ impl Timeline {
         Ok(())
     }
 
+    pub async fn edit_poll(
```

Comment:
Since the send api is `pub async fn send(&self, content: AnyMessageLikeEventContent)` could we do something similar for the edit one?
Changing 
```
pub async fn edit(
        &self,
        new_content: RoomMessageEventContent,
        edit_item: &EventTimelineItem,
    ) -> Result<(), UnsupportedEditItem>
```

Into something like this:
pub async fn edit(
        &self,
        new_content: AnyMessageLikeEventContent,
        edit_item: &EventTimelineItem,
    ) -> Result<(), UnsupportedEditItem>
```

It would make the API simpler from the client side.

Reply:
The semantics wouldn't change, if the even type doesn't support replies you'll still get  UnsupportedEditItem

Reply:
But I don't want this discussion to hinder the PR more as we're short on time... let's merge and worry about api shape later!